parameter/criterion/include/criterion/Criteria.h

```diff
@@ -0,0 +1,155 @@
+                                        core::log::Logger& logger,
+                                        std::string& error);
+
+    /** Criterion Retrieval
+     *
+     * @param[in] name the criterion name
+     * @result pointer to the desired criterion object
+     */
+    Criterion* getSelectionCriterion(const std::string& name);
+
+    /** Const Criterion Retrieval
+     *
+     * @param[in] name the criterion name
+     * @result pointer to the desired const criterion object
+     */
+    const Criterion* getSelectionCriterion(const std::string& name) const;
+
+    /** List available criteria
+     *
+     * @param[out] results information container
+     * @param[in] withTypeInfo indicates if we want to retrieve criterion type information
+     * @param[in] humanReadable indicates the formatage we want to use
+     */
+    void listSelectionCriteria(std::list<std::string>& results,
+                               bool withTypeInfo,
+                               bool humanReadable) const;
+
+    /** Reset the modified status of criterions */
+    void resetModifiedStatus();
+
+    /** Xml Serialization method
+     *
+     * @param[out] xmlElement the current xml element to fill with data
+     * @param serializingContext context of the current serialization
+     */
+    virtual void toXml(CXmlElement& xmlElement, CXmlSerializingContext& serializingContext) const;
+
+private:
+
+    /** Internally wrap criterion pointer to not have to handle destruction */
+    typedef std::unique_ptr<Criterion> CriterionWrapper;
+
+    /** Criteria instance container type, map which use criterion name as key */
+    typedef std::map<std::string, CriterionWrapper> CriteriaMap;
+
+    /** Confine exception use to smooth code transitions
+     * Android is not supporting exceptions by default. Nevertheless some
+     * solutions has emerged.
+     * This function allows to confine exceptions use in only one function.
+     * It also allows to easily connect this part of code to the parameter-framework
+     * which works without exceptions for now.
+     *
+     * @param[in] name, name of the criterion to retrieve
+     * @return pointer to the desired criterion
+     */
+    Criterion* getCriterionPointer(const std::string& name) const;
+
+    /**
+     *
+     */
+    template<class CriterionType>
+    Criterion* addCriterion(const std::string& name,
+                            const CriterionInterface::Values& values,
+                            core::log::Logger& logger,
+                            std::string& error)
+    {
+        try {
+            Criterion* criterion(new CriterionType(name, values, logger));
+            mCriteria.emplace(name, CriterionWrapper(criterion));
+            return criterion;
+        }
+        catch (Criterion::InvalidCriterionError& e) {
+            error = e.what();
+            return nullptr;
+        }
+    }
+
+    /** Criteria instance container */
+    CriteriaMap mCriteria;
```

Comment:
I suppose this class (Criteria) should be made non-copyable or `unique_ptr` replaced with `shared_ptr` in the definition of CriterionWrapper.